### PR TITLE
add the final repo

### PR DESCRIPTION
**Pull request by vinay156**
Added all function and readme file.
Review it !!

### CODE REVIEW

**Comment by vinay156** [comment]
`Description`
Endpoints

• Home:

    Home page -
    URI: GET /
    Status: 200
    Response: index.ejs

• Authentication:

    Sign Up -
      URI: GET /sign-up
      Status: 200
      Response: signup.ejs

      URI: POST /sign-up
      Status: 200
      Request: {
                    username,
                    password,
                }
      Response: redirect("/")

    Log In -
      URI: GET /log-in
      Status: 200
      Response: login.ejs

      URI: POST /log-in
      Request: {
                    username,
                    password,
                }
      Status: 200
      Response: redirect("/")

    Log Out -
      URI: GET /log-out
      Status: 200
      Response: redirect("/")

• Message:

    Create Message -
      URI: GET /create-message
      Status: 200
      Response: create-message.ejs

      URI: POST /create-message
      Request: {
                    title,
                    note,
                }
      Status: 200
      Response: redirect("/")

Database schema
User
username: String
password: String
status: member

Message
title: String
note: String
date: Date
user: User

virtual-
createdAt: Date

Tests
Automated test cases added
• None
Manual test cases run
• Creates sample test data
• Checks all the endpoints

Project Link:
https://membersonly789.herokuapp.com/

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+.env
```
Comment (coder1033) [inline .gitignore]: Add a newline.
Throughout all the files where it is missing.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+var createError = require('http-errors');
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
```
Comment (coder1033) [inline app.js]: Use `const` instead of `var`.
Make changes throughout all the files

File: app.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+const session = require("express-session");
+const passport = require("passport");
+const LocalStrategy = require("passport-local").Strategy;
+const mongoose = require('mongoose')
+const bcrypt = require('bcryptjs')
+const dotenv = require('dotenv');
```
Comment (coder1033) [inline app.js]: Require all of these alphabetically.

File: app.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+dotenv.config();
+
+
+//Mongoose connect
+mongoose.connect(process.env.DB_URL, { useUnifiedTopology: true, useNewUrlParser: true, useCreateIndex: true });
+const db = mongoose.connection;
+db.on("error", console.error.bind(console, "mongo connection error"));
+
+//Model
+const User = require('./models/user')
```
Comment (coder1033) [inline app.js]: Require everything on top of the file.

**Comment by coder1033** [comment]
Please use the prettier extension for code formatting.

File: controllers/message-controller.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+exports.createMessage = (req, res) => {
+    if(req.user){
+        res.render('create-message', {user: req.user})
+    }else{
+        res.redirect('/')
+    }
+}
+
+exports.createMessagePost = (req, res, next) => {
+        // const errors = validationResult(req)
```
Comment (coder1033) [inline controllers/message-controller.js]: Remove unnecessary comments.

File: routes/index.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    return next()
+    
+}
+
+router.get('/', (req, res) =>{
+    Message.find().exec((err, list_msg)=>{
+        if(err) {return next(err)}
+        res.render("index", {
+            user: req.user,
+            msg_list: list_msg
```
Comment (coder1033) [inline routes/index.js]: Use camel case. `msgList`

Comment (vinay156) [reply routes/index.js]: Updated

File: controllers/message-controller.js (hunk 5 above)
Comment (vinay156) [reply controllers/message-controller.js]: Updated

File: app.js (hunk 4 above)
Comment (vinay156) [reply app.js]: Updated

File: app.js (hunk 3 above)
Comment (vinay156) [reply app.js]: Updated

File: app.js (hunk 2 above)
Comment (vinay156) [reply app.js]: updated

File: .gitignore (hunk 1 above)
Comment (vinay156) [reply .gitignore]: Don't get it !!

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,3 @@
...
+.vscode
```
Comment (coder1033) [inline .gitignore]: Add new line at the end. Make this change in every file.

File: app.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+const cookieParser = require("cookie-parser");
+const dotenv = require("dotenv");
+const express = require("express");
+const session = require("express-session");
+const createError = require("http-errors");
+const mongoose = require("mongoose");
+const logger = require("morgan");
+const passport = require("passport");
+const LocalStrategy = require("passport-local").Strategy;
+const path = require("path");
```
Comment (coder1033) [inline app.js:11]: Import alphabetically as per the variable names

File: app.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+dotenv.config();
+
+//Model
```
Comment (coder1033) [inline app.js]: Remove these comments.

File: app.js [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+//EJS
+const app = express();
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "ejs");
+
+//Passport
+passport.use(
+  new LocalStrategy((username, password, done) => {
+    User.findOne({ username: username }, (err, user) => {
```
Comment (coder1033) [inline app.js]: `{ username }`

File: routes/index.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+function checkNotAuth(req, res, next) {
+  if (req.isAuthenticated()) {
+    res.redirect("/");
+  }
+  return next();
+}
+
+router.get("/", (req, res) => {
+  Message.find().exec((err, msgList) => {
```
Comment (coder1033) [inline routes/index.js]: Put this function into controller.

File: app.js (hunk 8 above)
Comment (vinay156) [reply app.js:11]: Updated

**Review by vinay156 — COMMENTED** [review]
Updated

File: app.js [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(session({ secret: "cats", resave: false, saveUninitialized: true }));
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(express.urlencoded({ extended: false }));
+app.use(express.static(path.join(__dirname, "public")));
+
+const indexRouter = require("./routes/index");
```
Comment (coder1033) [inline app.js]: `require` at the top of file. Please do look in other files too.

File: controllers/auth-controller.js [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+  Message.find().exec((err, msgList) => {
+    if (err) {
+      return next(err);
+    }
+    res.render("index", {
+      user: req.user,
+      msgList: msgList,
+    });
+  });
+};
```
Comment (coder1033) [inline controllers/auth-controller.js]: Is this a part of `auth-controller`?

File: routes/index.js [hunk 14]
```diff
@@ -0,0 +1,30 @@
+var express = require("express");
+var router = express.Router();
```
Comment (coder1033) [inline routes/index.js]: Assign this after all the `require` statements. Do look in other files too.

File: routes/index.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+router.post("/sign-up", authController.signUpPost);
+
+router.get("/log-in", checkNotAuth, authController.logIn);
+
+router.post("/log-in", authController.logInPost);
+
+router.get("/log-out", authController.logOut);
+
+router.get("/create-message", messageController.createMessage);
```
Comment (coder1033) [inline routes/index.js]: `'/message'` in both get and post.
Change the variable and function names also.

File: routes/index.js [hunk 16]
```diff
@@ -0,0 +1,30 @@
+var express = require("express");
+var router = express.Router();
+
+const authController = require("../controllers/auth-controller");
+const messageController = require("../controllers/message-controller");
+
+function checkNotAuth(req, res, next) {
+  if (req.isAuthenticated()) {
```
Comment (coder1033) [inline routes/index.js]: This should not be in this file.
Make a separate middleware directory for this.

File: app.js (hunk 9 above)
Comment (vinay156) [reply app.js]: Updated

File: app.js (hunk 12 above)
Comment (vinay156) [reply app.js]: Updated

File: controllers/auth-controller.js (hunk 13 above)
Comment (vinay156) [reply controllers/auth-controller.js]: Yes

File: routes/index.js (hunk 14 above)
Comment (vinay156) [reply routes/index.js]: Updated

File: app.js [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+const dotenv = require("dotenv");
+const express = require("express");
+const flash = require("connect-flash");
+const LocalStrategy = require("passport-local").Strategy;
+const logger = require("morgan");
+const mongoose = require("mongoose");
+const passport = require("passport");
+const path = require("path");
+const session = require("express-session");
+dotenv.config();
```
Comment (coder1033) [inline app.js]: Put these after all the require statements.

**Review by coder1033 — APPROVED** [review]

File: app.js (hunk 17 above)
Comment (vinay156) [reply app.js]: Updated

File: routes/index.js (hunk 16 above)
Comment (vinay156) [reply routes/index.js]: Done

**Pull request merged**